test/startup-script/manage-startup-script.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+CHECKPOINT_PATH="${CHECKPOINT_PATH:-/tmp/startup-script.kubernetes.io_$(md5sum <<<"${STARTUP_SCRIPT}" | cut -c-32)}"
+CHECK_INTERVAL_SECONDS="30"
+EXEC=(nsenter -t 1 -m -u -i -n -p --)
+
+do_startup_script() {
+  local err=0;
+
+  "${EXEC[@]}" bash -c "${STARTUP_SCRIPT}" && err=0 || err=$?
+  if [[ ${err} != 0 ]]; then
+    echo "!!! startup-script failed! exit code '${err}'" 1>&2
+    return 1
+  fi
+
+  "${EXEC[@]}" touch "${CHECKPOINT_PATH}"
+  echo "!!! startup-script succeeded!" 1>&2
+  return 0
+}
+
+while :; do
+  "${EXEC[@]}" stat "${CHECKPOINT_PATH}" > /dev/null 2>&1 && err=0 || err=$?
```

Comment:
Is `err=0 || err=$?` supposed to assign err if it isn't initialised? Or does it always make err = 0 but the expression is always 0 so it then does `err=$?`?
Either way, what is `$?` in this case? it's it always the exit code of sleep except for the first time?

Reply:
`$?` is the exit code of `"${EXEC[@]}" stat "${CHECKPOINT_PATH}" > /dev/null 2>&1` if it fails AFAIU. I think this line could be split into two lines like so:
```
"${EXEC[@]}" stat "${CHECKPOINT_PATH}" > /dev/null 2>&1
err=$?
```
And it would work the same, will fix it.

Reply:
After fixing it it failed the pod because we set `errexit`, and stat here is allowed to fail. I am reverting to an original implementation.